platform/ontology/api.go

```diff
@@ -71,71 +71,57 @@ func (p *Platform) CurrentBlockNumber() (int64, error) {
 	return height, nil
 }
 
-func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
+func (p *Platform) GetBlockByNumber(num int64) (block *blockatlas.Block, err error) {
 	response, err := p.client.GetBlockByNumber(num)
 	if err != nil {
-		logger.Error("GetBlockByNumber", logger.Params{"platform": p.Coin().Symbol, "details": err.Error()})
 		return nil, err
 	}
-	var (
-		block blockatlas.Block
-		txs   []blockatlas.Tx
-	)
 	if response.Error == 0 {
```

Comment:
can you verify the response error inside the client method and throw an error if it occurs?